base/set.jl

```diff
@@ -141,6 +141,27 @@ function unique(f::Callable, C)
     out
 end
 
+"""
+    alldistinct(itr)
+
+Test if all values from `itr` are distinct, when compared with `isequal`.
```

Comment:
@stevengj: thanks, pushed minor change correcting that.
@tkelman: Sorry that I did not branch, I am still learning the ropes, will do it next time.

I hope that this can be merged now, but let me know if I should do anything else.